CMakeLists.txt

```
@@ -76,11 +76,13 @@ if (NOT PYTHON_EXECUTABLE)
 endif()
 endif()
 
-find_hdf5()
-if (NOT HDF5_FOUND)
-	message(WARNING "hdf5 library not found, some tests will not be run")
-else()
-    include_directories(${HDF5_INCLUDE_DIR})
+if(BUILD_TESTS)
```

If you only include hdf5 iff you build tests, `flann_mpi_server` / `flann_mpi_client` from `src/cpp/CMakeLists.txt#L71` and `flann_example_cpp` / `flann_example_mpi` from `examples/CMakeLists.txt#L13` are never build unless you also build the tests.

```suggestion
if(USE_HDF5)
```
Something like this would follow the other options.

You'd also need to add this after L62:

```
option(USE_HDF5 "Use HDF5" OFF)
```

And this at the end.
```
message(STATUS "Using HDF5 support: ${USE_HDF5}")
message(STATUS "Using Parallel HDF5 support: ${HDF5_IS_PARALLEL}")
```

---

Or you use something like:
```suggestion
+if(BUILD_EXAMPLES OR BUILD_TESTS OR USE_MPI)
```